### PR TITLE
[Naitei][Admin][Ransack] Add search bar and filter status in bills list page

### PR DESCRIPTION
**Pull request by TranMinhHuy2003**
## Related Tickets
- [ticket redmine](https://edu-redmine.sun-asterisk.vn/issues/79590)

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
[Screencast from 2024-08-14 22-51-21.webm](https://github.com/user-attachments/assets/30c71310-cd92-4d4b-94a5-e3b1a9b5224a)
passrubocop_update_cart

### CODE REVIEW

**Comment by TranMinhHuy2003** [comment]
ready

File: app/views/admin/bills/_search_filter.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<div class="search-filter">
+  <%= form_tag admin_bills_path, method: :get, data: {controller: "bill"}, class: "search-filter-wrapper" do %>
+    <div class="search-wrapper">
+      <%= text_field_tag :search, params[:search], class: "form-control", placeholder: t("admin.view.bills.search_for"), data: {action: "change->bill#submit"} %>
+    </div>
+    <div class="filter-wrapper">
+      <h4 class="title"><%= t("admin.view.bills.status") %></h4>
```
Comment (ThaiKhacDuoc) [inline app/views/admin/bills/_search_filter.html.erb]:
```suggestion
      <h4 class="title"><%= t "admin.view.bills.status" %></h4>
```

File: app/views/admin/bills/show.html.erb [hunk 2]
```diff
@@ -22,6 +22,10 @@
       <th><%= t "admin.view.status" %></th>
       <td><%= @bill.status %></td>
     </tr>
+    <tr>
+      <th><%= t "admin.view.note" %></th>
+      <td><%= @bill.note_content %></td>
+    </tr>
     <tr>
       <th><%= t "admin.view.total" %></th>
       <td><%= format_currency(@bill.total) %></td>
```
Comment (ThaiKhacDuoc) [inline app/views/admin/bills/show.html.erb]: Thử này có chạy được k huynh
```suggestion
      <td><%= format_currency @bill.total %></td>
```

File: app/views/admin/bills/show.html.erb [hunk 3]
```diff
@@ -30,6 +34,10 @@
       <th><%= t "admin.view.total_after_discount" %></th>
       <td><%= format_currency(@bill.total_after_discount) %></td>
     </tr>
+    <tr>
+      <th><%= t "admin.view.cancel_reason" %></th>
+      <td><%= format_currency(@bill.cancellation_reason) %></td>
```
Comment (ThaiKhacDuoc) [inline app/views/admin/bills/show.html.erb]: Tương tự

**Review by ThaiKhacDuoc — COMMENTED** [review]
cmt

**Comment by TranMinhHuy2003** [comment]
fixed

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**